Add AI and poetry blog post (#382)
trinodb/trino.io: _posts/2022-12-06-poetry-ai.md

OpenAI project. So we had to try.

<!--more-->

Its pretty amazing that you can feed the AI a question we did:

```
Write a short rhyming poem explaining Trino in easy but accurate terms
```

And after a few trials, the best one produced by GPT-3 was this:

```
Trino is a tech platform,
That makes data integration a breeze,
It's designed to help your business grow,
And optimise your systems with ease.
It links together different sources,
Both big and small,
Gathers data of different types,
Helping you make decisions for all.
So Trino is an integration tool,
That connects data for you,
It'll ease the burden of data sync,
Giving you more time to do what you do.
```

Pretty amazing, huh? I am sure we could not have done better even after a long
time trying. If you are interested to learn more you can look at the [blog post
on Ars
Technica](https://arstechnica.com/information-technology/2022/11/openai-conquers-rhyming-poetry-with-new-gpt-3-update/)
or go and try it yourself [on the
playground](https://beta.openai.com/playground/).

Enjoy while we are heading back to [working on Trino pull
requests](https://github.com/trinodb/trino/pulls) and other code now.

> Macros and Martin

Reviewer: Typo: s/Macros/Marcos

Reviewer 2: Now we have a nickname for @mtraverso3

---
layout: post
title: "AI and poetry meet Trino"
author: Martin Traverso, Marcos Traverso
excerpt_separator: <!--more-->
image: /assets/trino-light.png
---

> Sometimes working in the depth of the query engine core, the planner, the

Reviewer 3: ```suggestion
Sometimes when working in the depth of the query engine core, the planner, the
```

optimizer, or some other tricky problem or code, you run out of steam and need a

> distraction. Your brain will quietly and in the background still work on the
> problem, while you get to read or do something interesting.

Reviewer 3: ```suggestion
distraction. Your brain will quietly work on the problem in the background
while you get to read or do something interesting.
```

That's when we found out more about GPT-3 and the recent improvements on the

> OpenAI project. So we had to try.

Reviewer 3: "That's when" feels a little jarring. Maybe "one of these times," or "during one break, we-" to make it more clear you're transitioning from talking about a general phenomenon to a specific story.

<!--more-->

> Its pretty amazing that you can feed the AI a request just like we did:

Reviewer 4: ```suggestion
It's pretty amazing that you can feed the AI a prompt just like we did:
```
Maybe use `prompt` instead of `request` as it is more commonly referred to as this.

```
Write a short rhyming poem explaining Trino in easy but accurate terms
```

And after a few trials, the best one produced by GPT-3 was this:

```
Trino is a tech platform,
That makes data integration a breeze,
It's designed to help your business grow,
And optimise your systems with ease.
It links together different sources,
Both big and small,
Gathers data of different types,
Helping you make decisions for all.
So Trino is an integration tool,
That connects data for you,
It'll ease the burden of data sync,
Giving you more time to do what you do.
```

Cool, huh? I am sure we could not have done better even after a long time

> trying. Have a look at another question:

Reviewer 4: ```suggestion
trying. Have a look at another prompt:
```
Same suggestion as above!

> trying. Have a look at another question:

Reviewer 3: This might be the bachelor's in English talking, but we definitely could've done better, and I'd probably rephrase this to be a little less grandiose of a claim...

```
Write a short rhyming poem explaining Trino, the SQL query engine in complex
and accurate terms.
```

And the compelling result:

```
Trino is a query engine that does more than just run,

> It’s an SQL engine with performance that stuns.

Reviewer 3: "an SQL"... the bot has chosen its side, and it has chosen wrong. 😆

It is secure, efficient, and kind,
Providing real-time analytics of any kind.
It’s a distributed system that’s built to scale,
Processing queries that never fail.
Connecting to myriad data sources,
It can quickly analyze any course.
It’s flexible, reliable, and fast,
And is designed to make your queries last.
It can handle large volumes of data with ease,
Making the most complex queries a breeze.
```

If you are interested to learn more you can look at the [blog post
on Ars

> Technica](https://arstechnica.com/information-technology/2022/11/openai-conquers-rhyming-poetry-with-new-gpt-3-update/)

Reviewer 3: I'd just squash the entire link text & link onto one line, it's more wide but also more readable that way.

or go and try it yourself [on the
playground](https://beta.openai.com/playground/).

Enjoy while we are heading back to [working on Trino pull
requests](https://github.com/trinodb/trino/pulls) and other code now.

> Macros and Martin

Reviewer 4: ```suggestion
Marcos and Martin
```
Typo
